packages/app/src/specs/generators/component/ReactComponentGeneratorStepOne.vue

```diff
@@ -0,0 +1,198 @@
+        <div
+          v-if="generateSpecMutation.fetching.value"
+          class="mt-48px w-full inline-flex items-center justify-center"
+        >
+          <i-cy-loading_x16 class="h-48px mr-12px animate-spin w-48px" />
+          <p class="text-lg">
+            Loading
+          </p>
+        </div>
+        <ExpandableFileChooser
+          v-else-if="!result"
+          v-model:extensionPattern="extensionPattern"
+          :files="allFiles"
+          :loading="query.fetching.value"
+          @select-item="makeSpec"
+        />
+        <GeneratorSuccess
+          v-else
+          :file="result.file"
+        />
+      </div>
+      <div>
+        <StandardModalFooter
+          v-if="result"
+          class="flex gap-16px items-center"
+        >
+          <router-link
+            class="outline-none"
+            :to="{ name: 'SpecRunner', query: { file: result.file.relative?.replace(/\\/g, '/') }, params: { shouldShowTroubleRenderingAlert: true } }
+            "
+          >
+            <Button
+              size="lg"
+              :prefix-icon="TestResultsIcon"
+              prefix-icon-class="w-16px h-16px icon-dark-white"
+              @click="emits('close')"
+            >
+              {{ t('createSpec.successPage.runSpecButton') }}
+            </Button>
+          </router-link>
+          <Button
+            size="lg"
+            :prefix-icon="PlusButtonIcon"
+            prefix-icon-class="w-16px h-16px icon-dark-gray-500"
+            variant="outline"
+            @click="emits('restart')"
+          >
+            {{ t('createSpec.successPage.createAnotherSpecButton') }}
+          </Button>
+        </StandardModalFooter>
+        <div
+          v-else
+          class="bg-white rounded-b h-24px bottom-0 left-0 absolute ghost-div"
+        />
+      </div>
+    </template>
+  </div>
+</template>
+<script setup lang="ts">
+import { useVModels, whenever } from '@vueuse/core'
+import { useI18n } from '@cy/i18n'
+import ExpandableFileChooser from '../ExpandableFileChooser.vue'
+import GeneratorSuccess from '../GeneratorSuccess.vue'
+import { computed, ref } from 'vue'
+import { gql, useQuery, useMutation } from '@urql/vue'
+import type { ReactComponentGeneratorStepOne_CodeGenGlobFragment, GeneratorSuccessFileFragment } from '../../../generated/graphql'
+import { ReactComponentGeneratorStepOneDocument, ReactComponentGeneratorStepOne_GenerateSpecDocument } from '../../../generated/graphql'
+import StandardModalFooter from '@cy/components/StandardModalFooter.vue'
+import Button from '@cy/components/Button.vue'
+import PlusButtonIcon from '~icons/cy/add-large_x16.svg'
+import TestResultsIcon from '~icons/cy/test-results_x24.svg'
+import EmptyGenerator from '../EmptyGenerator.vue'
+
+const props = defineProps<{
+  title: string
+  gql: ReactComponentGeneratorStepOne_CodeGenGlobFragment
+}>()
+const { t } = useI18n()
+const emits = defineEmits<{
+  (event: 'update:title', value: string): void
```

Comment:
Random q, why do we ever want to change the title/description? Did you find out? I have a feeling this is working around a flaw in the generator design, but I could be mistaken.

Reply:
It looks like if there's an error generating the spec file, we fall back to `EmptyGenerator` which updates the title once a spec is successfully generated 🤔 seems like there's probably a better way to do this

from `EmptyGenerator`
```typescript
whenever(result, () => {
  title.value = t('createSpec.successPage.header')
  emits('updateTitle', t('createSpec.successPage.header'))
})
```

Reply:
Looks like we duplicate some logic. Probably too much of a lift to work on that refactor here, but good to know and maybe something we can revisit...

Reply:
Thoughts on creating an issue to capture this?